text/0047-introduce-profile-context.md

```diff
@@ -0,0 +1,110 @@
+- Start Date: 2022-12-20
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/47
+- RFC Status: draft
+
+# Summary
+
+We want to introduce the concept of a profiling context, much like a tracing context, starting with
+transactions so we can easily relate a group of transactions back to the respective profile.
+
+# Motivation
+
+We already have a way to relate a group of profiles back to the respective transaction. And up to
+this point, we have been using this relationship to map single transaction at a time to the
+respective profile. This is now insufficient as we would like to be able to relate a group of
+transactions back to the respective profile. This is to enable various touch points into profiling
+in the existing performance product.
+
+The intended use cases include
+
+- fetching a list of transaction with the profile ID
+- filtering transactions to only those that have a profile
+
+# Background
+
+Profiles will be closely associated to transactions and creating this relation will enable us to
+build a richer integration between the Performance and Profiling products.
+
+Since the Profiling product will be initially tied to the Performance product, we want to ensure
+that there is a smooth integration through various touch points to allow users to navigate from
+Performance to Profiling and vice versa.
+
+# Options Considered
+
+## SDK
+
+### Option A: Add a new context type
+
+Introduce a key in the transaction payload under contexts called `profile` to include information
+related to the profile.
+
+```json
+{
+  ...,
+  "contexts": {
+    ...,
+    "profile": {
+      "profile_id": "<id>"
+    }
+  }
+}
+```
+
+### Option B: Use the existing trace context
+
+Use the existing trace context and add a new key for the `profile_id`.
+
+```json
+{
+  ...,
+  "contexts": {
+    ...,
+    "trace": {
+      "profile_id": "<id>"
+    }
+  }
+}
+```
+
+## Storage
+
+To support querying for this relationship, we have to store the profile ID, extracted from the
+profile context set by the SDK, on the existing transactions dataset. The types of queries we want
+to do are as follows:
+
+1. Selecting the profile ID
+2. Selecting rows that have a profile ID since not all transactions will have a profile
```

Comment:
So one common workflow that has come up is a user is looking at transactions in the performance product. Then they want to see a profile associated with it.

But because profiles can be sampled at a lower rate than transactions, not all transactions will have a profile associated with it. So we would like to provide a way for users to focus on transactions that **HAVE** a profile associated.

One *possible* flow for this is perhaps the user will type `has:profile_id` into the search bar to filter for just transactions that are known to have an associated profile.

Reply:
Thanks for the explanation. In the workflow you mentioned, since the user is interested in **any** transaction with a profile, I would recommend that you limit your search to a fixed number of transactions. There isn't really a need to search for all transactions within a time range for your use case.

Reply:
Yep understood.